### DISCUSSION TITLE
[✨] Nextjs style cache revalidation for server fetched data

### DISCUSSION THREAD

**Discussion by engineersamwell** [QwikDev/qwik-evolution#130] [Proposals For Qwik]
### Is your feature request related to a problem?

Currently when fetching data in a `routeLoader$` i've noticed stale data with back buttoning after changing an item.  Let's say I'm fetching products in a `routeLoader$` to be displayed on a products page, then I click a product that takes me to product 1 with name 'a' which I change the name to 'b' then I back button to the products page.  The list will still show product 1 with 'a' instead of the correct value 'b'.  If I forward and back again it will show the change.  

- Note this is with default settings and not changing the `cacheControl`.  I can fix this issue if I use `cacheControl` to disable the cache completely, but imo that's not optimal.  I should be able to aggressively cache content while having the manual control to revalidate that content only when I know it's changed.

### Describe the solution you'd like

Nextjs solves this issue with cache revalidation in a few ways, see: https://nextjs.org/docs/app/building-your-application/data-fetching/forms-and-mutations#revalidating-data

You can either revalidate a path:

```
export default async function submit() {
  await submitForm()
  revalidatePath('/')
}
```

Or revalidate a collection associated with a fetch:

```
export default async function Page() {
  const res = await fetch('https://...', { next: { tags: ['posts'] } })
  const data = await res.json()
  // ...
}

export default async function submit() {
  await addPost()
  revalidateTag('posts')
}
```

The `revalidateTag` is very precise and a solution I would love to see with Qwik.

### Describe alternatives you've considered

- One alternative is to disable caching or to use staleWhileRevalidate with a shorter `maxAge`
- Another would be to use client side fetching, but I don't really consider this an option as it negates much of the benefit of Qwik server side data fetching.

**Comment by gioboa** [comment]
We moved this issue to `qwik-evolution` repo to create a RFC discussion for this.
Here is our [Qwik RFC process](https://github.com/QwikDev/qwik-evolution?tab=readme-ov-file#qwik-rfc-process) thanks.
